Docker/docker-compose.yml

```diff
@@ -0,0 +1,147 @@
+version: "3.9"
```

Comment:
So I would have gone for the docker-compose replicas feature, which would have allowed you to have a much smaller docker-compose file, using port ranges to expose the ports on which to expose the ports of the validators.

Now I freely admit that I would probably have been an idiot to do that because your exact hardcoding of ports and their exposed variants in fact guarantees we will never have to deal with a port conflict in the next 2-5 weeks, which is the result we want to get to.

Can we nonetheless open an issue to track the need to configure this to more replicas than 4, after we're past our deadline?

Reply:
@huitseeker exactly as you said, that's why avoided to use the `replicas` feature. I am not fond of the long docker-compose file either and I was looking more into the direction of generating the compose yml file via a very simple python script accepting number of authorities & workers which would:
1.  generate the `authority keys` and write the necessary files
2. create the `committee.json file`
3. generate the `docker-compose` file

Alternatively we can have the script do 1 & 2 and configure things to avoid port conflict and make the docker-compose very small using the `replicas` future, but to be honest I like the simplicity and predicability of fixed common ports

Reply:
issue here https://github.com/MystenLabs/narwhal/issues/278